RecoTracker/LSTCore/src/alpaka/Quadruplet.h

```diff
@@ -0,0 +1,1065 @@
+      const int flatThreadExtent = blockSize;  // total threads per block
+
+      for (int iter : cms::alpakatools::uniform_groups_z(acc, nEligibleT4Modules)) {
+        uint16_t lowerModule1 = ranges.indicesOfEligibleT4Modules()[iter];
+
+        if (cms::alpakatools::once_per_block(acc)) {
+          matchCount = 0;
+        }
+
+        short layer2_adjustment;
+        short md_adjustment;
+        int layer = modules.layers()[lowerModule1];
+        if (layer == 1) {
+          if (modules.subdets()[lowerModule1] != Endcap)
+            continue;
+          layer2_adjustment = 1;
+          md_adjustment = 1;
+        }  // get upper segment to be in third layer
+        else if (layer == 2) {
+          if (modules.subdets()[lowerModule1] != Endcap)
+            continue;
+          layer2_adjustment = 1;
+          md_adjustment = 0;
+        }  // get lower segment to be in third layer
+        else {
+          layer2_adjustment = 0;
+          md_adjustment = 0;
+        }
+        unsigned int nInnerTriplets = tripletsOccupancy.nTriplets()[lowerModule1];
+
+        alpaka::syncBlockThreads(acc);
+
+        // Step 1: Make inner and outer triplet pairs
+        for (unsigned int innerTripletArrayIndex : cms::alpakatools::uniform_elements_y(acc, nInnerTriplets)) {
+          unsigned int innerTripletIndex = ranges.tripletModuleIndices()[lowerModule1] + innerTripletArrayIndex;
+          if (triplets.partOfPT5()[innerTripletIndex])
+            continue;  //don't create T4s for T3s accounted in pT5s
+          if (triplets.partOfT5()[innerTripletIndex])
+            continue;  //don't create T4s for T3s accounted in T5s
+          if (triplets.partOfPT3()[innerTripletIndex])
+            continue;  //don't create T4s for T3s accounted in pT3s
+          uint16_t lowerModule2 = triplets.lowerModuleIndices()[innerTripletIndex][1];
+          unsigned int nOuterTriplets = tripletsOccupancy.nTriplets()[lowerModule2];
+          for (unsigned int outerTripletArrayIndex : cms::alpakatools::uniform_elements_x(acc, nOuterTriplets)) {
+            unsigned int outerTripletIndex = ranges.tripletModuleIndices()[lowerModule2] + outerTripletArrayIndex;
+            if (triplets.partOfPT5()[outerTripletIndex])
+              continue;  //don't create T4s for T3s accounted in pT5s
+            if (triplets.partOfT5()[outerTripletIndex])
+              continue;  //don't create T4s for T3s accounted in T5s
+            if (triplets.partOfPT3()[outerTripletIndex])
+              continue;  //don't create T4s for T3s accounted in pT3s
+
+            unsigned int secondSegmentIndex = triplets.segmentIndices()[innerTripletIndex][1];
+            unsigned int thirdSegmentIndex =
+                triplets.segmentIndices()[outerTripletIndex][0];  //second and third segments are the same here
+            unsigned int innerOuterInnerMiniDoubletIndex =
+                segments.mdIndices()[secondSegmentIndex][0];  //inner triplet outer segment inner MD index
```

Comment:
please try to keep the names more compact or at least more readable: `innerOuterInner` requires the comment to be understood.

if the name was more clear the comment is not necessary
```suggestion
            unsigned int innerT3OuterLSInnerMDIndex = segments.mdIndices()[secondSegmentIndex][0];
```
or

```suggestion
            unsigned int innerT3MD2Index = segments.mdIndices()[secondSegmentIndex][0]; 
```